File: frontRecrutement/interviewFront-main/src/app/shared/classes/utilisateur.ts
import { Profile } from "./profile";
import { Test } from "./test";

export interface Utilisateur {
    /*
    @NotBlank
	private String nom;
	@NotBlank
	private String prenom;
	@Positive
	@Min(value = 18)
	
	private int age;
	@Email
	@NotBlank
	private String email;
	@NotBlank
	@Size(min = 6,max = 20)
	private String mdp;
	
	private Grades grade;
	@JsonIgnoreProperties("utilisateur")

	private ProfileDto profile;
	@JsonIgnoreProperties("utilisateurs")

	private List<TestDto> tests;

    */
    id: number;
    nom: string;
    prenom: string;
    age: number;
    email: string;
    motDePasse: string;
    grade: string;
    profile: Profile;
    tests: Test[];
    
    
}

File: backRecrutement/interviewApp-main/interviewApp-main/src/main/java/com/interview/model/entity/Utilisateur.java
//package com.interview.model.entity;
//
//import java.util.List;
//
//import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
//
//import jakarta.persistence.CascadeType;
//import jakarta.persistence.Column;
//import jakarta.persistence.Entity;
//import jakarta.persistence.EnumType;
//import jakarta.persistence.Enumerated;
//import jakarta.persistence.ManyToMany;
//import jakarta.persistence.OneToMany;
//import jakarta.persistence.OneToOne;
//import lombok.AllArgsConstructor;
//import lombok.Getter;
//import lombok.NoArgsConstructor;
//import lombok.Setter;
//
//@Entity
//@Getter
//@Setter
//@NoArgsConstructor
//@AllArgsConstructor
//
//public class Utilisateur  extends BaseEntity{
//
//@Column(nullable = false)
//private String nom;
//@Column(nullable = false)
//private String prenom;
//@Column(nullable = false,unique = true)
//private String email;
//@Column(nullable = false)
//private String mdp;
//@Enumerated(EnumType.STRING)
//private Grades grade;
//@OneToOne(mappedBy = "utilisateur")
//@JsonIgnoreProperties("utilisateur")
//
//private Profile profile;
//@ManyToMany(cascade = {
//		CascadeType.PERSIST,
//		CascadeType.MERGE
//})
//private List<Test> tests;
//@OneToMany(mappedBy = "emetteur")
//@JsonIgnoreProperties("emetteur")
//private List<Candidature> candidatures;
//@OneToMany(mappedBy = "emetteur")
//@JsonIgnoreProperties("emetteur")
//private List<Message> envoyees;
//@OneToMany(mappedBy = "recepteur")
//@JsonIgnoreProperties("recepteur")
//private List<Message> recus;
//@OneToMany(mappedBy = "recruteur")
//@JsonIgnoreProperties("recruteur")
//private List<Offre> offres;
//
//
//}
